app/models/ems_refresh/save_inventory_container.rb

```diff
@@ -382,6 +383,26 @@ def save_labels_inventory(entity, hashes, target = nil)
     store_ids_for_new_records(entity.labels, hashes, [:section, :name])
   end
 
+  def save_labels_and_tags_inventory(entity, hashes, target = nil)
+    save_labels_inventory(entity, hashes, target)
+    save_tags_generated_from_labels(entity, hashes, target)
+  end
+
+  # Must be called after object already got its labels.
+  # This is achieved by :labels_and_tags going through save_labels_and_tags_inventory().
+  # TODO: Drop the order requirement by using the :labels hash instead of entity.labels?
+  #   Would still need still save_labels_and_tags_inventory() to get that hash.
+  def save_tags_generated_from_labels(entity, _hashes, _target = nil)
+    current_tags = ContainerLabelTagMapping.tags_for_entity(entity)
+    possible_old_tags = ContainerLabelTagMapping.all_mapped_tags
+
+    # TODO: Don't clear existing tags if partial refresh?
+    #       Check `target` like all other methods do.
```

Comment:
I think you can drop this comment since we do not support partial refresh. Even if we do in the future, if you got to this method, deleting all tags for entity is what you want to do, since you are refreshing THIS entity, no?